src/mbgl/style/style.cpp

```diff
@@ -240,6 +251,26 @@ Source* Style::getSource(const std::string& id) const {
     return it != sources.end() ? it->get() : nullptr;
 }
 
+std::vector<std::string> Style::getAttributions() const {
+    std::vector<std::string> result;
+    result.reserve(sources.size());
```

Comment:
To clarify, this will always allocated more memory than we need since we always have a annotation source that doesn't have attribution.